### PR TITLE
Add unit tests for profiler module

### LINKED ISSUE #226: Add basic, structural unit tests for the profiler

### PR DESCRIPTION
**Pull request by HarshaSatyavardhan**
Closes #226                                                                        
- Adds 20 unit tests for `src/mmore/profiler.py`
- Tests configuration (`configure_profiling`, `get_profiling_config`, `enable_profiling_from_env`)                                                       
- Tests timing utilities (`time_function`, `time_context`)                         
- Tests profiling utilities (`profile_function`, `profile_context`, `Profiler` class)                                                                             
- Covers edge cases: `sys.getprofile` guard, `profile_functions=False`, disabled mode, stop-without-start safety

### CODE REVIEW

File: tests/test_profiler.py [hunk 1]
```diff
@@ -0,0 +1,309 @@
...
+    time_function,
+)
+
+
+@pytest.fixture(autouse=True)
+def reset_profiler_config(tmp_path):
+    """Reset global config before/after each test using tmp_path to avoid repo pollution."""
+    configure_profiling(enabled=False, output_dir=str(tmp_path / "default_profiling"))
+    yield
+    configure_profiling(enabled=False, output_dir=str(tmp_path / "default_profiling"))
```
Comment (JCHAVEROT) [inline tests/test_profiler.py]: I don't think it's really useful to reset before the test, you always overwrite the default profiler configuration in your tests


```suggestion
@pytest.fixture(autouse=True)
def reset_profiler_config(tmp_path):
    """Reset global config after each test."""
    yield
    configure_profiling(enabled=False, output_dir=str(tmp_path))
```

File: tests/test_profiler.py [hunk 2]
```diff
@@ -0,0 +1,309 @@
...
+    with patch.dict(
+        os.environ,
+        {
+            "MMORE_PROFILING_ENABLED": "true",
+            "MMORE_PROFILING_OUTPUT_DIR": env_dir,
+            "MMORE_PROFILING_SORT_BY": "calls",
+            "MMORE_PROFILING_MAX_RESULTS": "25",
+        },
+        clear=False,
+    ):
```
Comment (JCHAVEROT) [inline tests/test_profiler.py]: Cleaner to use [monkeypatch](https://docs.pytest.org/en/stable/how-to/monkeypatch.html), additionally you already use the fixture in `test_enable_profiling_from_env_defaults()`

```suggestion
    monkeypatch.setenv("MMORE_PROFILING_ENABLED", "true")
    monkeypatch.setenv("MMORE_PROFILING_OUTPUT_DIR", env_dir)
    monkeypatch.setenv("MMORE_PROFILING_SORT_BY", "calls")
    monkeypatch.setenv("MMORE_PROFILING_MAX_RESULTS", "25")
```

File: tests/test_profiler.py [hunk 3]
```diff
@@ -0,0 +1,309 @@
...
+    env_vars_to_remove = [
+        "MMORE_PROFILING_ENABLED",
+        "MMORE_PROFILING_OUTPUT_DIR",
+        "MMORE_PROFILING_SORT_BY",
+        "MMORE_PROFILING_MAX_RESULTS",
+    ]
+    with patch.dict(os.environ, {}, clear=False):
+        for k in env_vars_to_remove:
+            os.environ.pop(k, None)
+        enable_profiling_from_env()
```
Comment (JCHAVEROT) [inline tests/test_profiler.py]: Here also you can leverage the `monkeypatch` fixture


```suggestion
   for k in (
        "MMORE_PROFILING_ENABLED",
        "MMORE_PROFILING_OUTPUT_DIR",
        "MMORE_PROFILING_SORT_BY",
        "MMORE_PROFILING_MAX_RESULTS",
    ):
        monkeypatch.delenv(k, raising=False)

    enable_profiling_from_env()
```

File: tests/test_profiler.py [hunk 4]
```diff
@@ -0,0 +1,309 @@
...
+@patch("mmore.profiler.logger")
+def test_time_function_with_parens(mock_logger):
+    """@time_function() with parens returns correct result and logs."""
+
+    @time_function(log=True)
+    def add(a, b):
+        return a + b
+
+    result = add(2, 3)
+    assert result == 5
```
Comment (JCHAVEROT) [inline tests/test_profiler.py]:
```suggestion
    assert add(2, 3) == 5
```

File: tests/test_profiler.py [hunk 5]
```diff
@@ -0,0 +1,309 @@
...
+@patch("mmore.profiler.logger")
+def test_time_function_no_parens(mock_logger):
+    """@time_function without parens also works correctly."""
+
+    @time_function
+    def multiply(a, b):
+        return a * b
+
+    result = multiply(3, 4)
+    assert result == 12
```
Comment (JCHAVEROT) [inline tests/test_profiler.py]:
```suggestion
    assert multiply(3, 4) == 12
```

File: tests/test_profiler.py [hunk 6]
```diff
@@ -0,0 +1,309 @@
...
+@patch("mmore.profiler.logger")
+def test_time_function_no_log(mock_logger):
+    """@time_function(log=False) suppresses logger calls."""
+
+    @time_function(log=False)
+    def noop():
+        return "done"
+
+    result = noop()
+    assert result == "done"
```
Comment (JCHAVEROT) [inline tests/test_profiler.py]:
```suggestion
    assert noop() == "done"
```

File: tests/test_profiler.py [hunk 7]
```diff
@@ -0,0 +1,309 @@
...
+def test_profile_function_disabled(tmp_path):
+    """When disabled, no .prof file is written and return value is correct."""
+    configure_profiling(enabled=False, output_dir=str(tmp_path))
+
+    @profile_function()
+    def greet():
+        return "hello"
+
+    result = greet()
+    assert result == "hello"
```
Comment (JCHAVEROT) [inline tests/test_profiler.py]:
```suggestion
    assert greet() == "hello"
```

File: tests/test_profiler.py [hunk 8]
```diff
@@ -0,0 +1,309 @@
...
+def test_profile_function_enabled(_mock_getprofile, _mock_print_stats, tmp_path):
+    """When enabled, a .prof file is written with func name in filename."""
+    configure_profiling(enabled=True, output_dir=str(tmp_path))
+
+    @profile_function()
+    def compute():
+        return 42
+
+    result = compute()
+    assert result == 42
```
Comment (JCHAVEROT) [inline tests/test_profiler.py]:
```suggestion
    assert compute() == 42
```

File: tests/test_profiler.py [hunk 9]
```diff
@@ -0,0 +1,309 @@
...
+def test_profile_function_profile_functions_false(tmp_path):
+    """enabled=True but profile_functions=False → no .prof file written."""
+    configure_profiling(enabled=True, output_dir=str(tmp_path), profile_functions=False)
+
+    @profile_function()
+    def task():
+        return "result"
+
+    result = task()
+    assert result == "result"
```
Comment (JCHAVEROT) [inline tests/test_profiler.py]:
```suggestion
    assert task() == "result"
```

**Review by JCHAVEROT — COMMENTED** [review]
Hi @HarshaSatyavardhan thanks for the contribution once again 👍

I left a few small remarks, feel free to take a look.
Also it could be great if you take a look at removing some tests that are a bit redundant, I think we can make things a bit simpler. And last thing It could be a really nice addition if you add a few tests to check the actual content of the files generated by the Profiler (for example if a profiled function is called 10 times inside a loop in one of your test, you check that `total_calls == 10` in the output file -- you can dig a bit more inside `profiler.py` to see which other output would be worth to test)

**Review by JCHAVEROT — APPROVED** [review]
Thanks for making the changes, I took a look to the tests you added to check the actual content of the profiler results and they look good 👍
